eras/babbage/impl/src/Cardano/Ledger/Babbage/Translation.hs

```diff
@@ -58,23 +59,25 @@ instance TranslateEra BabbageEra NewEpochState where
         , stashedAVVMAddresses = ()
         }
 
-instance TranslateEra BabbageEra Tx where
-  type TranslationError BabbageEra Tx = DecoderError
-  translateEra _ctxt tx = do
-    -- Note that this does not preserve the hidden bytes field of the transaction.
-    -- This is under the premise that this is irrelevant for TxInBlocks, which are
-    -- not transmitted as contiguous chunks.
-    txBody <- translateEraThroughCBOR "TxBody" $ tx ^. bodyTxL
-    txWits <- translateEraThroughCBOR "TxWitness" $ tx ^. witsTxL
-    auxData <- case tx ^. auxDataTxL of
-      SNothing -> pure SNothing
-      SJust auxData -> SJust <$> translateEraThroughCBOR "AuxData" auxData
-    let validating = tx ^. isValidTxL
-    pure $
-      mkBasicTx txBody
-        & witsTxL .~ txWits
-        & auxDataTxL .~ auxData
-        & isValidTxL .~ validating
+instance Typeable l => TranslateEra BabbageEra (Tx l) where
+  type TranslationError BabbageEra (Tx l) = DecoderError
+  translateEra _ctxt tx =
+    withTopTxLevelOnly tx $ \tx' ->
+      do
```

Comment:
This doesn't fit the style we use everywhere else. Let's not put `do` on its own line, unless we really have to.
```suggestion
    withTopTxLevelOnly tx $ \tx' -> do
```